R/tables.R

```diff
@@ -1,117 +1,274 @@
+ed_table_ <- function(table, ..., .dots) {
+  dots <- lazyeval::all_dots(.dots, ...)
+  dots = lazyeval::as.lazy_dots(   #This stuff deals with the dplyr bug found at https://github.com/hadley/dplyr/issues/511 by modifying "%in%" calls
+    lapply(dots, function(dot_expr) {
+      new_expr <- paste0(
+        deparse(partial_eval(dot_expr[["expr"]], env=dot_expr[["env"]]),
+                width.cutoff = 500L),
+        collapse = "")
+      if(stri_detect_fixed(new_expr, "%in%")) {
+        matched_expr <- stri_extract_last_regex(new_expr, "(?<=%in%\\s).*$")
+        if(length(eval(parse(text=matched_expr))) ==  0 ) {
+          new_expr <- stri_replace_first_fixed(new_expr, matched_expr, "('')")
+        }
+        else if(length(eval(parse(text=matched_expr))) == 1) {
+          new_expr <- stri_replace_first_fixed(new_expr, matched_expr,
+                                               paste0("(", matched_expr, ")"))
+        }
+      }
+      lazyeval::as.lazy(new_expr, env=dot_expr[["env"]])
+    }))
+
+
+  #add class attributes to P2 tables
+  if(stri_detect_fixed(table, "2")){
+    ed_tb <- dbReadTable(eidith_db()$con, table)
+    #adding notes
+    note_cols <- which(stri_detect_fixed(names(ed_tb), "notes"))
+    attr(ed_tb, "notes") <- note_cols
+    #adding duplicate rows
+    intended_key <- db_other_indexes[[table]][[1]]
+    group_var <- as.name(intended_key)
+    key_errors <- ed_tb %>%
+      group_by(!!group_var) %>%
+      summarize(count = n()) %>%
+      filter(count > 1) %>%
+      dplyr::pull(!!group_var)
+    attr(ed_tb, "duplicate_keys") <- key_errors
+    indices <- which(ed_tb[[intended_key]] == key_errors)
+    attr(ed_tb, "duplicate_indices") <- indices
+    class(ed_tb) <- c("eidith_tbl", class(ed_tb))
+
+    #print details on local database pull"
+    if(interactive()){
+    if(length(note_cols > 0)){
+      cat_line("")
+      cat_line(cyan("There are notes attached to this dataframe which may contain important information! Please look at the following column(s):"))
+      cat_line(paste("    ", magenta(names(ed_tb)[note_cols])))
+    }
+    if(length(key_errors > 0)){
+      cat_line("")
+      cat_line(red("IMPORTANT: There are multiple rows with duplicate unique ID's in this table!"))
+      cat_line(black("              The duplicate IDs are:"))
+      cat_line(paste("                  ", red(duplicate_keys)))
+      cat_line(black("              This affects the following rows:"))
+      cat_line(paste("                  ", red(duplicate_indices)))
+    }
+  }
+  }else{
+    ed_tb <- tbl(eidith_db(), table)
+    ed_tb %>%
+      filter_(.dots=dots) %>%
+      collect(n=Inf) %>%
+      fix_classes()
+  }
+ed_tb
+}
+
+
+
+
+# Add mention of attributes on pull call
+
+#'@importFrom stringr str_detect
+#'@importFrom cli cat_line
+#'@importFrom crayon cyan magenta
+#'@importFrom tibble as.tibble
+#'@export
+print.eidith_tbl <- function(x,...){
+  note_cols <- attributes(x)$notes
+  print(as.tibble(unclass(x)))

```

Comment:
This should pass on the `...` arguments so that you can, for instance, set the number of rows to print with `n=`.

Reply:
Added.